.github/workflows/sync-to-k8s.yml

```diff
@@ -0,0 +1,94 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+name: Sync feature PRs to kafka-k8s-operator
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  sync-pr:
+    runs-on: ubuntu-latest
+    # Run for feature branches only
+    if: contains(github.head_ref, 'dpe-')
```

Comment:
Maybe this can be refined a bit more, but we can always close wrong PRs on the k8s repo